src/screens/CreateNFT.js

```diff
@@ -0,0 +1,333 @@
+import ModalSendTx from '../components/ModalSendTx';
+import ModalAlert from '../components/ModalAlert';
+import { connect } from "react-redux";
+import BackButton from '../components/BackButton';
+import hathorLib from '@hathor/wallet-lib';
+import helpers from '../utils/helpers';
+import { NFT_STANDARD_RFC_URL } from '../constants';
+import InputNumber from '../components/InputNumber';
+
+
+const mapStateToProps = (state) => {
+  const HTR_UID = hathorLib.constants.HATHOR_TOKEN_CONFIG.uid;
+  const htrBalance = HTR_UID in state.tokensBalance ? state.tokensBalance[HTR_UID].available : 0;
+  return {
+    htrBalance,
+    wallet: state.wallet,
+  };
+};
+
+
+/**
+ * Create an NFT
+ *
+ * @memberof Screens
+ */
+class CreateNFT extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.address = React.createRef();
+    this.inputWrapper = React.createRef();
+
+    /**
+     * errorMessage {string} Message to show when error happens on the form
+     * name {string} Name of the created token
+     * configurationString {string} Configuration string of the created token
+     * amount {number} Amount of tokens to create
+     */
+    this.state = {
+      errorMessage: '',
+      name: '',
+      configurationString: '',
+      amount: null,
+    };
+  }
+
+  /**
+   * Validates if the create NFT form is valid
+   */
+  formValid = () => {
+    const isValid = this.refs.formCreateNFT.checkValidity();
+    if (isValid) {
+      if (this.refs.address.value === '' && !this.refs.autoselectAddress.checked) {
+        this.setState({ errorMessage: t`Must choose an address or auto select` });
+        return false;
+      }
+
+      // Validating maximum amount
+      if (this.state.amount > hathorLib.constants.MAX_OUTPUT_VALUE) {
+        this.setState({ errorMessage: t`Maximum NFT units to mint is ${hathorLib.constants.MAX_OUTPUT_VALUE}` });
+        return false;
+      }
+      return true;
+    } else {
+      this.refs.formCreateNFT.classList.add('was-validated')
+    }
+  }
+
+  /**
+   * Opens PIN modal if form is valid
+   */
+  onClickCreate = () => {
+    if (!this.formValid()) {
+      return;
+    }
+
+    this.setState({ errorMessage: '' });
+    $('#pinModal').modal('show');
+  }
+
+  /**
+   * Prepare create NFT transaction data after PIN is validated
+   *
+   * @param {String} pin PIN written by the user
+   *
+   * @return {hathorLib.SendTransaction} SendTransaction object
+   */
+  prepareSendTransaction = async (pin) => {
+    // Get the address to send the created tokens
+    let address = '';
+    if (this.refs.autoselectAddress.checked) {
+      address = hathorLib.wallet.getAddressToUse();
+    } else {
+      address = this.refs.address.value;
+    }
+
+    const name = this.refs.shortName.value;
+    const symbol = this.refs.symbol.value;
+    const nftData = this.refs.nftData.value;
+    const createMint = this.refs.createMintAuthority.checked;
+    const createMelt = this.refs.createMeltAuthority.checked;
```

Comment:
`refs` was [deprecated](https://reactjs.org/docs/refs-and-the-dom.html#legacy-api-string-refs).

Reply:
Refactor done at https://github.com/HathorNetwork/hathor-wallet/pull/222/commits/e8bccac102b5d57f7bc6f864d29d7390e8f4d69e